lib/apiservers/engine/backends/image.go

```diff
@@ -179,3 +193,76 @@ func convertV1ImageToDockerImage(image *metadata.ImageConfig) *types.Image {
 		Labels:      labels,
 	}
 }
+
+// Retrieve the image metadata from the storage port layer.
+func getImageConfigFromCache(image string) (*metadata.ImageConfig, error) {
+	// Use docker reference code to validate the id
+
+	digest, named, err := reference.ParseIDOrReference(image)
+	if err != nil {
+		return nil, err
+	}
+
+	// Try to get the image config from our image cache
+	imageCache := ImageCache()
+
+	if digest != "" {
+		config := imageCache.GetImageByDigest(digest)
+		if config != nil {
+			return config, nil
+		}
+	} else {
+		config := imageCache.GetImageByNamed(named)
+		if config != nil {
+			return config, nil
+		}
+	}
+
+	return nil, derr.NewRequestNotFoundError(fmt.Errorf("No such image: %s", image))
+}
+
+// Converts the data structure retrieved from the portlayer.  This src datastructure
+// represents the unmarshalled data saved in the storage port layer.  The return
+// data is what the Docker CLI understand and returns to user.
+func imageConfigToDockerImageInspect(imageConfig *metadata.ImageConfig, productName string) *types.ImageInspect {
```

Comment:
This should probably have some validation in it to make sure all of the fields of the `imageConfig` struct are 1) set and 2) valid, no?  I would check for the fields being set and return an error if they aren't there sane.